R/compiler.R

```diff
@@ -137,6 +137,12 @@ SubstraitCompiler <- R6::R6Class(
     #'
     #' @return `self`
     validate = function() {
+
+      # DuckDB backend doesn't accept empty SELECT clause
+      if (inherits(self, "DuckDBSubstraitCompiler") && length(self$schema$names) == 0) {
+        rlang::abort("Column list must not be empty")
+      }
```

Comment:
Ah yeah, of course it belongs there; cheers!  

Out of interest, why does it need `super$validate()`? Is that in case we ever put other stuff in the parent class that we want to do before we call the DuckDB stuff?

And, done!

Reply:
Nothing now, but cool, I see why we would!